src/config/config_manager.cc

```diff
@@ -1007,14 +1007,14 @@ void ConfigManager::addOption(config_option_t option, std::shared_ptr<ConfigOpti
 
 void ConfigManager::load(const fs::path& userHome)
 {
-    std::string temp;
+    fs::path temp;
```

Comment:
`temp` is reused in several places, some as string some as path. You need to introduce a separate variable `tempPath`.